### DISCUSSION TITLE
Community pages should have a way that users can request to join that community

### DISCUSSION THREAD

**Discussion by jeremyjs** [knowledgefutures/pubpub#1535] [Ideas]
It can just be a "request to join community" button that would show up on the community page.

Only logged in users who are not members should be able to see this button.

The admin of the community should be able to toggle a setting to turn this feature off. In other words, there should be a new community setting that configures whether this community is open to requests to join.

This would require an additional admin-only page for review of pending requests, to approve or hide/deny each request, and a notification feature for admins (they would toggle an account+community setting to configure whether they shoudl receive an email, etc.) whenever someone requests to join the community.

Finally, this feature could also be extended to toggle communities to be one step more public, i.e. admins could toggle a setting so that anyone can join a community and participate at the press of the join button without any extra approval flow needed.

**Comment by gabestein** [comment]
Thanks so much! This is a really cool idea, and potentially not too tough to implement. It also strikes me as similar to [this comment](https://github.com/pubpub/pubpub/discussions/1323#discussioncomment-533773) about allowing for new ways of engagement.

We'll discuss at our next dev call!
